actionpack/lib/action_controller/metal/data_streaming.rb

```diff
@@ -155,7 +156,17 @@ def send_file_headers!(options)
         disposition = options.fetch(:disposition, DEFAULT_SEND_FILE_DISPOSITION)
         unless disposition.nil?
           disposition  = disposition.to_s
-          disposition += %(; filename="#{options[:filename]}") if options[:filename]
+          filename = options[:filename]
+          if filename
+            disposition += %(; filename="#{filename}")
+
+            # Support RFC 6266
+            # http://tools.ietf.org/html/rfc6266#appendix-D
+            unless filename.ascii_only?
+              encoded_filename = CGI::escape(filename)
+              disposition += %(; filename*="#{encoded_filename.encoding}''#{encoded_filename}")
```

Comment:
@icambron Note we use 5987 for `filename*` only and do %-escape there. We use an old-style `quoted-string` for the ASCII `filename`, where spaces are allowed, for wide compat with older browsers.